### ISSUE TITLE
Can't call scala.StringBuilder#length_= in 2.13.0-M5

### ISSUE THREAD

**Issue by xuwei-k** [scala/bug#11158]
https://github.com/scala/scala/blob/v2.13.0-M5/src/library/scala/collection/mutable/StringBuilder.scala#L68

```
Welcome to Scala 2.13.0-M5 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> val s = new StringBuilder
s: StringBuilder =

scala> s.length = 42
                ^
       error: reassignment to val

scala> s.length_=(42)
```

```
Welcome to Scala 2.12.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> val s = new StringBuilder
s: StringBuilder =

scala> s.length = 42
s.length: Int = 42
```

🤔

**Comment by SethTisue** [comment]
what's the underlying cause?

**Comment by xuwei-k** [comment]
minimal reproduce example:


### CharSeqJava.java

```java
package example;

public interface CharSeqJava {
  int length();
}
```

### Main.scala

```scala
package example

trait CharSeqScala {
  def length: Int
}

class A extends CharSeqJava {
  override def length: Int = 1
  def length_=(value: Int): Unit = {}
}

class B extends CharSeqScala {
  def length: Int = 1
  def length_=(value: Int): Unit = {}
}

class C {
  def length: Int = 1
  def length_=(value: Int): Unit = {}
}

class D extends CharSeqJava with CharSeqScala {
  def length: Int = 1
  def length_=(value: Int): Unit = {}
}

class E extends CharSeqScala with CharSeqJava {
  def length: Int = 1
  def length_=(value: Int): Unit = {}
}

object Main {
  val a = new A
  val b = new B
  val c = new C
  val d = new D
  val e = new E

  // compile error
  a.length = 1

  // success
  b.length = 1

  // success
  c.length = 1

  // success
  d.length = 1

  // compile error
  e.length = 1 
}
```

**Comment by xuwei-k** [comment]
Should we swap mixin order? 🤔 

- https://github.com/scala/scala/blob/v2.12.6/src/library/scala/collection/mutable/StringBuilder.scala#L31-L37
- https://github.com/scala/scala/blob/v2.13.0-M5/src/library/scala/collection/mutable/StringBuilder.scala#L30-L34

```diff
 final class StringBuilder(val underlying: java.lang.StringBuilder) extends AbstractSeq[Char]
+  with java.lang.CharSequence 
   with Builder[Char, String]
   with IndexedSeq[Char]
-  with IndexedSeqOps[Char, IndexedSeq, StringBuilder]
+  with IndexedSeqOps[Char, IndexedSeq, StringBuilder] {
-  with java.lang.CharSequence {
```

**Comment by Jasper-M** [comment]
I think we should.

**Comment by adriaanm** [comment]
This goes back to scala/scala@b739c4a -- since we're "overriding" a Java method with an empty argument list, we add one to the `length` method, which then disqualifies it from being a getter (`treeInfo.mayBeVarGetter` expects a `NullaryMethodType`, which has no argument lists)

**Comment by SethTisue** [comment]
@adriaanm tentatively moved to RC1 milestone, you can make the call though

**Comment by adriaanm** [comment]
I won't have time for a proper fix. It's kind of tricky :-/

**Comment by SethTisue** [comment]
not sure what PR fixed it, but it's been working since 2.13.2

**Comment by som-snytt** [comment]
> what's the underlying cause?
```
@inline def length: Int = underlying.length
```
> not sure what PR fixed it, but it's been working since 2.13.2

https://github.com/scala/scala/pull/8558

which instead of fixing getter detection, fixes setter detection for assignment rewrite.

Regression test at https://github.com/scala/scala/pull/10357

Worth adding that there are a few contexts now in which Lukas has said recently (since 2.13) it's time to revert adding a param list to an override of Java (`toString()`) instead of just handling the override correctly.
